llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -3012,8 +3015,7 @@ void DwarfDebug::emitDebugARanges() {
   for (auto &I : SectionMap) {
     MCSection *Section = I.first;
     SmallVector<SymbolCU, 8> &List = I.second;
-    if (List.size() < 1)
-      continue;
+    assert(!List.empty());
```

Comment:
After convincing myself that any `List` must be non-empty, I was confident that the early `return` could be placed above this loop. I replaced the dead check with an assert to avoid similar head scratching in future.

Reply:
I believe the assert is appropriate even without the early return. If ArangeLabels is empty, SectionMap will also be empty, and the loop body will not execute.

Reply:
Yep, agreed. Just wanted to explain why I touched this seemingly unrelated part, for posterity.